tools/ports/__init__.py

```diff
@@ -57,13 +55,32 @@ def load_port(name):
   if not hasattr(port, 'variants'):
     # port variants (default: no variants)
     port.variants = {}
+  if not hasattr(port, 'show'):
+    port.show = lambda: f'{port.name} (--use-port={port.name}; {port.license()})'
 
   for variant, extra_settings in port.variants.items():
     if variant in port_variants:
       utils.exit_with_error('duplicate port variant: %s' % variant)
     port_variants[variant] = (port.name, extra_settings)
 
 
+def validate_port(port):
+  expected_attrs = ['get', 'clear', 'show']
+  if port.is_contrib:
+    expected_attrs += ['url', 'description', 'license']
+  for a in expected_attrs:
+    assert hasattr(port, a), 'port %s is missing %s' % (port, a)
+
+
+# Called from test_sanity
+def validate_ports():
```

Comment:
One reason I thought we might want to do this at runtime is that we might want to allow folks to drop in new ports (or maybe as you suggest using URL of a port on the command line).  In that case it might be good to validate each time you use a port.  However, we can address that once we get to that point.

Reply:
Of course if/when ports are added dynamically or loaded on demand this would have to change.

Reply:
I mean technically you can do today by simply dropping a new file into `tools/ports`.. but I'm not sure anyone is doing that.

Reply:
I reverted the change and now `validate_ports()` is called every time ports are read (line 99) for the potential cases where somebody might drop a file in `tools/ports`